projects/igniteui-angular/src/lib/grids/grid/grid-row-selection.spec.ts

```diff
@@ -1671,6 +1673,40 @@ describe('IgxGrid - Row Selection #grid', () => {
             expect(grid.getRowByIndex(1).selected).toBeTruthy();
             expect(grid.onRowSelectionChange.emit).toHaveBeenCalledTimes(6);
         }));
+
+        it('Should select only filtered records', async() => {
```

Comment:
Try to change the test to be fakeAsync and remove the grid.cdr.detectChanges() calls if possible.

Reply:
Test seems to work without grid detectChanges, so just removing them.